fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/indexes/RankIndexMaintainerFactory.java

```diff
@@ -64,15 +65,25 @@ public void validate(@Nonnull MetaDataValidator metaDataValidator) {
 
             @Override
             public void validateChangedOptions(@Nonnull Index oldIndex, @Nonnull Set<String> changedOptions) {
+                // Allow changing from unspecified to the default (or vice versa), but not otherwise.
                 if (changedOptions.contains(IndexOptions.RANK_NLEVELS)) {
-                    int oldLevels = RankIndexMaintainer.getNLevels(oldIndex);
-                    int newLevels = RankIndexMaintainer.getNLevels(index);
+                    int oldLevels = RankedSetIndexHelper.getNLevels(oldIndex);
+                    int newLevels = RankedSetIndexHelper.getNLevels(index);
                     if (oldLevels != newLevels) {
                         throw new MetaDataException("rank levels changed",
                                 LogMessageKeys.INDEX_NAME, index.getName());
                     }
                     changedOptions.remove(IndexOptions.RANK_NLEVELS);
                 }
+                if (changedOptions.contains(IndexOptions.RANK_HASH_FUNCTION)) {
+                    RankedSet.HashFunction oldFunction = RankedSetIndexHelper.getHashFunction(oldIndex);
+                    RankedSet.HashFunction newFunction = RankedSetIndexHelper.getHashFunction(index);
+                    if (!oldFunction.equals(newFunction)) {
+                        throw new MetaDataException("rank hash function changed",
+                                LogMessageKeys.INDEX_NAME, index.getName());
```

Comment:
Interesting. I suppose we could loosen this to allow for the rank hash function to change (given that it should be compatible). I'm not opposed to being stricter for now, though perhaps that should be an option at some point.

Reply:
Yes. I thought being conservative during the transition was best.